linters/trufflehog/trufflehog_to_sarif.py

```diff
@@ -28,6 +28,33 @@ def to_result_sarif(path: str, line_number: int, vuln_id: str, description: str)
     }
 
 
+secret_line_cache = {}
+file_cache = {}
+
+
+def find_line_number(secret, path):
+    if path not in file_cache:
+        file_cache[path] = open(path).readlines()
+
+    if secret not in secret_line_cache:
+        secret_line_cache[secret] = []
+
+    secret_length = len(secret.splitlines())
+    lines = file_cache[path]
+
+    for lineno in range(1, len(lines) - secret_length):
+        check_window = lines[lineno : lineno + secret_length]
```

Comment:
let's lift this out into a helper, `sliding_window`, c.f. itertools example https://docs.python.org/3/library/itertools.html:

```
def sliding_window(iterable, n):
    # sliding_window('ABCDEFG', 4) --> ABCD BCDE CDEF DEFG
    it = iter(iterable)
    window = collections.deque(islice(it, n-1), maxlen=n)
    for x in it:
        window.append(x)
        yield tuple(window)
```